org.scala-ide.sdt.core/src/org/scalaide/core/IScalaProject.scala

```diff
@@ -151,7 +153,7 @@ trait IScalaProject extends Publisher[IScalaProjectEvent] {
    * Initialization for the build manager associated to this project
    * @return an initialized EclipseBuildManager
    */
-  def buildManager(): EclipseBuildManager
+  def buildManager(): BuildScopeUnitsManager
```

Comment:
I've just fixed it. The EclipseBuildManager has been extended to enclose these cases where action on manager instance depended on its implementation.